source/usage-examples/transaction-core.txt

```diff
@@ -0,0 +1,233 @@
+  their past orders
+- Documents in the ``inventory`` collection that include quantities and
+  descriptions of all items
+
+The following document is in the ``customers`` collection:
+
+.. code-block:: json
+
+   { _id: 98765, orders: [] }
+
+The ``inventory`` collection contains the following documents:
+
+.. code-block:: json
+
+   { item: "sunblock", item_id: 5432, qty: 85 },
+   { item: "beach towel", item_id: 7865, qty: 41 }
+
+You store purchase records in the ``orders`` collection of the
+``testdb`` database. This collection is empty, as there have been no
+purchases.
+
+The code examples use the ``cart`` and ``payment`` variables to represent
+a sample list of items purchased and the order payment details. The
+following code describes the contents of the ``cart`` and ``payment`` variables:
+
+.. code-block:: javascript
+
+   const cart = [
+     { item: 'sunblock', item_id: 5432, qty: 1, price: 5.19 },
+     { item: 'beach towel', item_id: 7865, qty: 2, price: 15.99 }
+   ];
+   const payment = { customer: 98765, total: 37.17 };
+
+.. _nodejs-transaction-core-api-example:
+
+Implementation
+~~~~~~~~~~~~~~
+
+The code example in this section demonstrates how to use the Core API to
+perform a multi-document transaction in a session. In this example, the
+transaction makes the changes needed when a customer purchases items from
+your shop.
+
+This example code performs a transaction through the following actions:
+
+1. Calls the ``startSession()`` method to create a new session
+
+#. Calls the ``startTransaction()`` method with an options parameter to
+   create a new transaction
+
+#. Performs the following operations within the transaction:
+
+   * Inserts a document to the ``orders`` collection that contains
+     information about the purchase and customer
+   * Updates the ``inventory`` collection if there is
+     sufficient inventory to fulfill the purchase
+   * Ends the transaction and throws an exception if there isn't
+     sufficient inventory for any item in the order
+   * Adds the ID of the order to the list of past orders for the customer
+   * Returns a message acknowledging that the transaction
+     committed successfully with a copy of the purchase record
+
+#. Calls the ``commitTransaction()`` method to commit the transaction if
+   all operations complete successfully
+
+#. Implements a ``catch`` block that contains error-handling logic
+
+#. Calls the ``abortTransaction()`` method to terminate the transaction
+
+#. Calls the ``endSession()`` method to terminate the session
+
+.. note::
+
+   You can use this example to connect to an instance of MongoDB
+   and interact with a database that contains sample data. To learn more
+   about connecting to your MongoDB instance, see the :doc:`Usage Examples
+   guide </usage-examples>`.
+
+.. literalinclude:: /code-snippets/transactions/txn-core.js
+   :language: javascript
+   :linenos:
+   :emphasize-lines: 3-7,10,13,24,38,51,55,58-67,71,74
```

Comment:
Issue:
These seem to be off by a line in certain places (blank and nonfunctional lines are highlighted). 

Suggestion:
I think it could be good to only highlight items that need specific callouts, if any. The convenient transaction API usage example in this PR doesn't include any emphasize-lines attribute.

Reply:
I'll just remove the highlighting from the UEs and leave in the transaction examples.